### DISCUSSION TITLE
When will System.CommandLine package become GA

### DISCUSSION THREAD

**Discussion by samaniehsan** [dotnet/command-line-api#2139] [General]
Hi,
I noticed that System.CommandLine on nuget and https://learn.microsoft.com/en-us/dotnet/standard/commandline/
have been marked as preview for almost a year now.

what is the plan to mark these packages as GA?

I was hoping to use its capability to detect shutdown signals on k8s.

**Comment by shaggygi** [comment]
Yeah, it has been some time since and would like to see it reach GA.  I've had a few projects broken/rewritten a few times based on changes overtime (problem with cutting edge 😄).  Team has been working hard and believe it will be worth it once released. 

Might want to keep an eye out for https://github.com/dotnet/command-line-api/issues/2071

Also came across this.  https://www.youtube.com/watch?v=yDQGsZSEDOk

**Comment by KathleenDollard** [comment]
We are working on some design changes due to API review and hope to have an announcement on this quite soon.

**Reply by mesakomarevich** [reply]
That would be fantastic, as I am in the same boat as @shaggygi

**Reply by epsmae** [reply]
@KathleenDollard any update on that? This is probably the preview package with the most downloads ever. The latest preview version has 6.5M Downloads and is already used in many production applications. Wouldn't it make sense to mark it stable?

**Reply by DamianReeves** [reply]
Its really been silent on this front and it would be great to know what is the future of the project.

**Reply by elgonzo** [reply]
@epsmae @DamianReeves 

This released today/yesterday: https://github.com/dotnet/command-line-api/issues/2338
